tests/test_cli.py

```diff
@@ -1,4 +1,5 @@
 import pytest
+from control.server import GatewayServer
```

Comment:
Why is this needed? Can't we mock it up to keep unit-testing isolated?

Reply:
Thank you for the helpful remarks and review @epuertat 🖖, 

I see value in system integration tests, I do not mind that approach.

By performing system integration tests without extensive mocking, we can gain confidence in the overall functionality and behavior of the system, ensuring that the components interact correctly and produce the desired outcomes. It's important to strike a balance between unit tests and system integration tests. Both have their place in a comprehensive testing strategy, and each serves different purposes. 

WDYT?

Reply:
Yup, I think we should have both: unit-testing, and integration testing. Probably we should store each type of test in a different subdir to make it more clear: `tests/unit` and `test/integration`.

For integration tests, I'd usually prefer to reproduce a user workflow, rather than iteratively test individual commands, but we can do both. 

Main issue with integration tests is that they usually leave a trail of 'state' behind, so to perform integration testing safely, one usually has to implement teardown/clean-up logic to get rid of the state left by each previous test (and many times the best/only way to ensure reproducible tests is to recreate the testing environment from scratch).

BTW, Ceph itself has its own integration testing framework ([teuthology](https://docs.ceph.com/projects/teuthology/en/latest/README.html)), but that's not yet integrated with Github CI and requires multiple steps & waiting for resource allocation, so I'd start running some basic sanity tests with Github Actions, and then we can explore a more comprehensive integration testing with teuthology (btw, if I'm not wrong @oritwas was having a look at that).